include/zephyr/bluetooth/bluetooth.h

```diff
@@ -1817,8 +1817,11 @@ int bt_le_per_adv_sync_delete(struct bt_le_per_adv_sync *per_adv_sync);
  * such as synced, terminated and when data is received.
  *
  * @param cb Callback struct. Must point to memory that remains valid.
+ *
+ * @retval 0 Success.
+ * @retval -ENOENT if @p cb was already registered.
```

Comment:
Why `-ENOENT`? 

```
       ENOENT          No such file or directory (POSIX.1-2001).

                       Typically, this error results when a specified pathname does not exist, or one of the  com‐
                       ponents  in the directory prefix of a pathname does not exist, or the specified pathname is
                       a dangling symbolic link.
```

Not sure how that fits here to be honest. 
`-EALREADY`? `-EEXISTS`? 

I would also suggest to add tests of these new return values if possible (just as unit tests or something)

Reply:
I can update that. Do you have a sample unit test that can be taken as a starting point? There seems like there are no unit tests for the host that can be used directly